### ISSUE TITLE
Technical improvements of the rankings

### ISSUE THREAD

**Issue by AlbertoPdRF** [thewca/worldcubeassociation.org#4439]
After #4394 is merged, some of the code there should be improved. In particular, instead of using raw SQL for the queries, we should try to use ActiveRecord. At the moment, there are 4 different queries:

1. "N persons": I believe using the RanksSingle/Average tables would be easier for this option. The problem I encountered is that those tables don't include the `competitionId`. Maybe this could be resolved by adding the `competitionId` to the tables?
2. "N results": there are actually two different queries for this option, depending on if we want to show N single or average results. For the average option, simply getting the result should be enough. I'm not sure about the single one, but probably that would be more than enough too!
3. "By region": I think in this case we could as well just get the result.

**Comment by Baiqiang** [comment]
I don't think we can use RanksSingle for "N persons of some country/continent" because some competitors have changed their countries. They should show on "N persons" for their previous countries but their rankings are based on the current countries.

**Comment by jonatanklosko** [comment]
Another point: we could set up caching for rankings. The PHP cache is cleared as [a part of CAD](https://github.com/thewca/worldcubeassociation.org/blob/81d852de3ff33e2df075729275f71db8281a78e9/WcaOnRails/lib/auxiliary_data_computation.rb#L130-L133), so in Rails we could use CAD timestamp in cache key.

**Comment by maconard** [comment]
Maybe I didn't catch this somewhere, but it seems the "All" option is gone for rankings and you can only get 100/1000. This was intentional? (For both Person and Result.)

![Screenshot_20190815-084621~2](https://user-images.githubusercontent.com/31703304/63098862-77797380-bf39-11e9-91c9-542e6fad020f.png)

**Comment by AlbertoPdRF** [comment]
The "All persons" option was not working anymore with the old PHP code, so I just got rid of it. Adding pagination as @jonatanklosko suggested [here](https://github.com/thewca/worldcubeassociation.org/issues/4479#issuecomment-521385566) is an option to have it back! (And there wasn't an "All results" option.)

As a quick workaround: the number on the URL can be modified to show any arbitrary number of persons/results.
